Koowolab/private dns public preview

kay-owolabi opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

https://github.com/Azure/azure-rest-api-specs-pr/pull/156
https://github.com/Azure/azure-rest-api-specs-pr/pull/136

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
<details><summary>Swagger to SDK encountered a Subprocess error: (Azure/azure-sdk-for-go)</summary><p>

Command: profileBuilder -s preview -name preview
Finished with return code 127
and output:
```shell
/bin/sh: 1: profileBuilder: not found
```

</p></details>

lmazuel commented on specification/dns/resource-manager/readme.md:
```diff
@@ -29,6 +29,15 @@ openapi-type: arm
 tag: package-2017-09
```
Update this default package as well to package-2018-03-preview

bitsforthought committed:
[Azure Search] Adding examples to latest data plane specs (#2472)

* [Azure Search] Adding examples to latest data plane specs

Note that the Azure Search data plane API docs are not currently generated
from Swagger, but we plan to enable this eventually. These examples are just
to satisfy the Swagger validator in the meantime.

* [Azure Search] Fixing typos in synonyms examples

felixwa committed:
Revise Notice (#2492)

Revise Notice after PMs confirmed with CELA.

lmazuel committed:
Fix BatchAI readme (#2493)

vrdmr committed:
Adding Examples and fixing the DSC related lint issues (#2415)

VitaliyKurokhtin committed:
Redeclared consumed and produced MIME types supported in addition to application/json for /apis/{id} for GET/PUT operations (#2491)

* byRequest path fixed, schema reference fixed

* /reports/by... routes fixed

* Declared content types supported by API in addition to application/json

* Fixed get to produce mime types

djyou committed:
[EventGrid/ACR] Add schema of ACR events published to Event Grid (#2487)

* Add event grid schema for ACR

* Update readme file

* Prefix class name with ContainerRegistry

JinFengMS committed:
Update service map swagger spec to latest and fix validation errors (#2231)

* fix service map swagger validation errors

* remove the extra comma in examples/Machines/MachineGroups/SMMachinesListMachineGroupMembershipGet.json

* remove extra comma for client groups operation examples

* ARM operation examples: replace parameter workspaceId with workspaceName

* add required parameter machineGroupName for example SMMachineGroupsUpdatePut.json

* fix resource id and types

* fix more example validation errors

* fix validation errors

vrdmr committed:
Fixing the resource definitions (#2496)

lmazuel committed:
Move automation py to 0.2.0 (#2498)

VitaliyKurokhtin committed:
Redeclared consumed and produced MIME types supported in addition to application/json for /policies/{id} for GET/PUT operations (#2495)

* byRequest path fixed, schema reference fixed

* /reports/by... routes fixed

* Declared content types supported by API in addition to application/json

* Fixed get to produce mime types

* Redeclared consumed and produced MIME types supported in addition to application/json for /policies/{id} for GET/PUT operations

mcardosos committed:
Corrected API versions (#2482)

* Corrected API versions

* Move to correct folder

* Fix version

* Correct swagger version

* Fixed example

Commit:
Fail Breaking Changes CI on exception (#2500)

lmazuel commented:
@AutorestCI rebuild

nkindberg commented:
@lmazuel  FYI do not merge this PR until sign-off from either myself or karravi-msft. It is not ready to go. There are issues here we have been addressing with Kay offline.

AutorestCI commented:
# Automation for azure-sdk-for-go

lmazuel commented:
@nkindberg Thanks, I'm labelling this PR as do not merge until you tell it's fine.

AutorestCI commented:
<details><summary>Encountered a Subprocess error</summary><p>

Command: profileBuilder -s list -l ./profiles/2017-03-09/defintion.txt -name 2017-03-09
Finished with return code 127
and output:
```shell
/bin/sh: 1: profileBuilder: not found
```

</p></details>

lmazuel commented:
(@nkindberg I take the opportunity of this PR to do some automation tests, sorry for the notitications this might impact for a few minutes :/)

lmazuel commented:
@AutorestCI rebuild azure-sdk-for-python

lmazuel committed:
HDInsight Python conf (#2501)

solankisamir committed:
Remove duplicate operations (#2459)

Tag_ operations have all been moved to apimtags.json file. Removing them from apimproducts.json

Apis_ListByTags have been moved to apimapis.json, removing the file apimapisbytags.json

marstr committed:
Update base-branch for Swagger -> SDK tooling. (#2507)

ms-premp committed:
Updating $filter parameter sample values in examples. Earlier version had incorrect format for the date values. (#2499)

* Refactoring Pricesheet response spec to fix model names and references.

* Updating examples to correctly reflect the response model

* Fixing examples for updated pricesheet model response.

* Providing nextLink name and removing x-ms-pageable for budget PUT operation.

* Fixing nextLink definition in response schema.

* Updating $filter parameter sample values in examples. Earlier version had incorrect format for the date values.

Commit:
[AZURE] Microsoft.AzureStack namespace specs (#2484)

* AzureStack

* Formatted and Fixed some casings

AnatoliB committed:
Fix Automation swagger spec errors in the AutomationAccount and Webhook areas (#2360)

* Make 'properties' not required for PATCH on AutomationAccount_Update

* Make 'name' not required for PATCH on 'Webhook_Update'

* Add x-ms-azure-resource=true to Webhook definition

Commit:
Modify availableStacks model (#2506)

jovillas committed:
Add disk details to storage profile (#2458)

Make possible to provide more detailed information about the storage
devices attached to a given hana instance.

jhendrixMSFT committed:
Change statusMessage property to wildcard type. (#2486)

The statusMessage can be a string or a JSON object, so model it as a
wildcard type.

akshaysngupta committed:
appgw swagger examples (#2503)

* appgw examples #2

* ssl options example

* waf examples

Commit:
Microsoft.Automation linkedWorkspace operations (#2490)

* Microsoft.Automation linkedWorkspace operations

* Microsoft.Automation LinkedWorkspace operations use $ref ResourceGroupNameParameter

lmazuel committed:
Update monitor Py conf (#2512)

VitaliyKurokhtin committed:
Finally added missing policy mime types in all places (#2510)

avneeshraiusit committed:
Adding new API version 2018-01-10 fixing swagger violations. (#2419)

* Adding new API version 2018-01-10 fixing swagger voilations

* Updating readme file for new API:2018-01-10

* Updating api examples

* Reverting alphabetical ordering of operations and definitions for easier review.

* Taking review comments and introducing missing polymorphic fields for NetworkMappingFabricSpecificSettings

* Alphabetically sorting definitions and API paths.

haricr committed:
[DataFactory] Add new features to Swagger (#2504)

* Add new features
1. SecureOutput and BatchCount
2. Filter Activity
3. LInked Service Parameters
4. FilterActivity

* Fix indentations

* Fix review comments

mcardosos committed:
Added custom key identifier (#2403)

bitsforthought committed:
[Azure Search] Splitting data plane Swagger specs into separate direc… (#2515)

This seems like the path of least resistance to getting AutoRest (and the
scripts in azure-sdk-for-net that invoke it) to properly handle a data plane
SDK with two specs. Also, we're planning to split our data plane into separate
libraries, so this is a sensible first step.

LarryZhang19 committed:
[DO NOT MERGE, until feb 20] Adding 2018-01-01 version for metrics API (#2435)

* Adding 2018-01-01 version for metrics API

* Fix extra comma at the end of line

* Fix PR comments 1

* Fixing top/orderby names, Please be reminded, there is no code change, it's just a typo
of the names.

* Fix the sample to reflect the parameter

* Fix meta data sample as well.

irrogozh committed:
Updated network watcher swagger and examples for 2018-01-01 version (#2508)

* Updated examples for network watcher

* Updated api version in network watcher swagger and examples

* Fixed Linter errors

* Fixed Linter errors: iteration 2

dragav committed:
adding purge protection vault property (#2478)

* adding purge protection vault property

* Expanding the description of the soft delete and purge protection properties, respectively.
Updating examples with and without soft delete and purge protection flags.

* relocating the newly introduced property to the end of the schema

darrelmiller committed:
Add API Management Version Sets (#2483)

* Added ApiVersionSets

* Fixes based on CR

* Refactored JSON Schemas for payloads

* fixed names and reference

* Fixed reference to schema for ApiVersionSets

* Fixed ApiVersionSetCreate example

darshanhs90 committed:
fix the hardcoding of api version for classicadmin in powershell (#2502)

* fix the hardcoding of api version for classicadmin in sdk

* fix 2017 version preview file

najams committed:
AzAutomation Swagger for job & runbook  (#2466)

* Added swagger spec for job

* fixed unintentional update to
softwareUpdateConfigurationCollectionItemProperties

* Examples

* Fixed job schema & examples

* Fixed description and schema

* Fixed runbook and test job and their examples.

* Added x-ms-mutability to job and testJob

* Fixed build failures

* Addressed review comments: 1. make properties read-only 2. Define enum for provisioningState

* Fixed examples

promoisha committed:
Diagnostics entity added to APIs (#2497)

yuelilu33 committed:
Support logicAppReceivers in actionGroup apis (#2511)

jpiyali committed:
Add swagger for annotation and work item config api for microsoft.insights components namespace for api version 2015-05-01 (#2384)

* Add swagger for annotation api for microsoft.insights components namespace for api version 2015-05-01

* Address pr comment to include file in readme

* Fix issues with PR=true

* Fix more PR=true errors

* fix more errors

* Fix more PR=true errors

* Fix syntax error

* Fix more errors

* Fix syntax

* fix 2 more Pr=true errors

* Try to fix errors again

* Fix syntax

* Fix more errors

* Add work item configuration api

* Fix pr=true error

* Address code review comment for replacing error codes with defaultError definition

* Remove error codes from examples

naveedaz committed:
Suppress RPC R3019 per discussion at https://github.com/Azure/azure-r… (#2423)

* Suppress RPC R3019 per discussion at https://github.com/Azure/azure-rest-api-specs/pull/2258

* Add per path suppression for RPC 3019

Commit:
Relay: Added the missing API discovered in ARM_RP_Fundamentals (#2468)

* Added the missing API discovered in ARM_RP_Fundamentals

* Fix Python typo

* indentation fixes

lmazuel committed:
locationbasedservices py conf (#2533)

alexeldeib committed:
Add purge operations to AppInsights management plane (#2528)

* Add purge to management plane API

* Adding purge status operation

* Change purge statue to enum

* capitalize response status

alexeldeib committed:
[AppInsights Data Plane] Fix table schema (#2527)

* Remove internal schena column

* Fix row schema

d3r3kk committed:
Application Insights: Add Favorites and Web Test Locations APIs (#2416)

* Initial addition of favorites and locations API

* Complete web test locations API

- favor 'web test' names/descriptions over 'synthetic monitor' as web test is more customer-centric naming
- create example data for the web test api

* Spelling update

* Updated parameters for favorite API

* Add favorites API and examples

- ensure tests run against the applicationinsights folder successfully

* Add collectionFormat to comma-separated tags list parameter for favorites
- generating API in CSharp generated errors

* Enum for favorite type should be set to modelAsString as further favorite types are possible in future.

* Update suppressions to include specific instances I need suppressed (no global suppressions where possible).

* Suppress some noisey test warnings from API specifications from my peers.

* PR suggestions

- Remove x-ms-pageable annotation from our list operation, in the service code we never paganate our data.
- Add x-ms-enum annotation to one enum data type where it was missing.

Commit:
Event Hub: Added missing API to respective API version Swaggers (#2457)

* Added the missing API discovered in ARM_RP_Fundamentals to the respective API version swagger and review comments

* added tag for 2014-09

areddish committed:
Update the output-folder to match azure-sdk-for-net paths in Vision package (#2529)

Commit:
Added enableBatchedOperations property to ServiceBus Queue (#2534)

QingqingYuan committed:
[Don't Merge, Service roll out 2/28/2018] Swagger For new rdbms API change for new biz model (#2414)

* Add new version for mysql and pgsql

* Correct readme version

* Correct create properties parameter

* Correct PG namespace and some other comment

* Update Examples folder to examples

* Correct property name cases in the examples

* Corrected the ServerResult objects in CRUD samples

* Correct property values for server response in samples

* make createtime and lastmodifiedtime for logfileproperties to readonly

lmazuel committed:
Web Python conf (#2549)

amarzavery committed:
Add a directive for node and TS (#2550)

bulentelmaci committed:
Swagger specs for all existing versions of Microsoft.PolicyInsights (#2523)

* Swagger specs for all existing versions of Microsoft.PolicyInsights

* Add support for --validate; make where clauses explicit in existing (approval request pending) suppressions; add new suppression for equivalent paths error (approval request pending)

* Fix $top in examples; fix casing in examples

* Fix two more instances of $top type in examples

* Fix metadata examples by adding content type header and changing the body to a single string

* Remove EQUIVALENT_PATH suppressions; add extendedProperties as additionalProperties; add @odata.count to operations; remove "type" from "rows" in first 2017-08-09-preview and add descriptions

* Fix remaining examples with bad casings

* Fix additionalParameters definition; fix duplicate paths

* Remove redundant operations path/definitions from one of the swagger files for each version (one per namespace is sufficient as long as both files are fed into autorest as a bundle)

lmazuel committed:
DataMigration py conf (#2553)

michimune committed:
Fix recommendation resource properties, add disable recommendation APIs (#2548)

wilcobmsft committed:
Add Billing Account resource to Billing RP. (#2517)

* Add "Classic Account" ARM resource.

* 1. Change classic account property name from "userPrincipalName" to "principalName"
2. Add examples

* Fix API version in examples.

* Add example links.

* ClassicAccount->BillingAccount

* Removing "uuid" - billing account name may be a concatenation of uuid and additional keys.

* Remove unused parameters.

* 1. Add new version to readme.md
2. Fix validation warnings

* Fix descriptions.

* Fix type mismatch error.

* Adding missing parameter in example.

* Fix ms-examples name.

priydes committed:
Compute spec change for zone resilient images (#2551)

* Compute spec change for zone resilient images

* Address feedback from the previous iteration

payiAzure committed:
[DO NOT MERGE]update swaggers for azure elastic pool operation (#2520)

* update swaggers for azure database and elastic pool operation

* update ListDatabaseOperations and ListElasticPoolOperations examples

* Only update the swagger for elastic pool operations

* end

* remove two not needed folders. Fix the error in ListElasticPoolOperation

* Update the cancel operations swagger on database and pool, and corresponding examples

* fix indentation in cancelOperations.json and cancelPoolOperations.json

hitenjava committed:
Update Azure Database Migration Service to add Managed Instance as target  (#2531)

* Copy over 2017-11-15-preview folder in preparation for adding 2018-03-15-preview api version.

* Update Azure Database Migration Service to add Managed Instance as target

QingqingYuan committed:
Take out GeoRestore option (#2559)

gucalder committed:
[Monitor] Fixing issue with metricsnamespace incorrecly specified as client level argument #2565 (#2566)

* [Monitor] Make all attributes of EventData class readonly (#2254)

* [Monitor] Fixing issue with metricnamespace specification as client level argument #2565

bitsforthought committed:
[Azure Search] Change output path for .NET data plane SDK (#2554)

This is part of a planned refactoring that will split the Azure Search .NET
data plane SDK into separate packages.

Commit:
Fix links in the PR template (#2571)

jovillas committed:
Remove unused API (#2564)

Please note this is not a GA API and has been deprecated for a while.
There is no current user for it.

marstr committed:
Adding preview/stable to the example layout. (#2561)

* Adding preview/stable to the example layout.

* Fixing formatting

jhendrixMSFT committed:
fixed Go SDK path for automation (#2562)

pkamath13 committed:
[Monitor] Fixing issue with metrics api queryparameter name (#2575) (#2577)

najian committed:
add databricks activity (#2568)

SiddharthChatrolaMs committed:
Swagger completeness for Redis Cache (#2537)

* Default string modeled as parameter for auto generation to work correctly

* Adding CheckNameAvailability and listUpgradeNotifications into swagger for redis

* Fixing code review comments

* remove "/" at end for list apis

* Name change for notification

* Fixing code review comments in PR

* Fixing examples that are breaking the build

jhendrixMSFT committed:
Add missing type specifier to enum definition. (#2580)

bulentelmaci committed:
Change group name for operations path, and rename response type definition (#2563)

* Swagger specs for all existing versions of Microsoft.PolicyInsights

* Add support for --validate; make where clauses explicit in existing (approval request pending) suppressions; add new suppression for equivalent paths error (approval request pending)

* Fix $top in examples; fix casing in examples

* Fix two more instances of $top type in examples

* Fix metadata examples by adding content type header and changing the body to a single string

* Remove EQUIVALENT_PATH suppressions; add extendedProperties as additionalProperties; add @odata.count to operations; remove "type" from "rows" in first 2017-08-09-preview and add descriptions

* Fix remaining examples with bad casings

* Fix additionalParameters definition; fix duplicate paths

* Remove redundant operations path/definitions from one of the swagger files for each version (one per namespace is sufficient as long as both files are fed into autorest as a bundle)

* Change group name for operations path, and rename response type definition

* Change group name for operations path, and rename response type definition (old API versions)

solankisamir committed:
ApiManagement 2018-01-01 version spec (#2292)

* Copy the 2017-03-01 files

* 2018-01-01 specific changes

revert x-ms-enum and update description

* 202 on create or update, network status and backend flatten

202 on create or update, networkstatus backend flatten

Head for ApiRevision

* Fix ValidationRegex of Ids and length

* Fix the TagResource contract

* Make PUT operation on portalsettings resources ARM-compliant

Fix examples

* Removed x-ms-client-flatten

* NextLinkName and fix Check Entity Exist

* Bring back default responses and fix examples

Commit:
Copy 2017-10-01 DNS API version to 2018-03-01-preview
